epub33/rs/index.html

```diff
@@ -1852,33 +1867,73 @@ <h4>Scripting Considerations</h4>
-							approach will isolate publications from each other, thereby limiting access to cookies, DOM
-							storage, etc.</p>
-						<div class="note">
-							<p>Although domain isolation is the most effective way to secure EPUB Publications in
-								Web-based Reading Systems, it is not always a realistic option. Web-based Reading
-								Systems must still maintain isolation in such cases, which may require limiting access
-								to scripting APIs that allow client-side data storage and access.</p>
-						</div>
-						<p>Reading Systems that enable scripting and network access should also include methods to
-							notify the user that network activity is occurring and/or that allow them to disable it.</p>
+						<p>
+							Reading Systems must ensure that the data tied to specific <a href="#confreq-rs-scripted-origin">origins</a>
+							(e.g., Web Storage [[?WEBSTORAGE]] or IndexedDB [[?INDEXEDDB]]) is preserved between "reading".
+							Reading System implementations that integrate the capabilities of web browser engines (e.g., embedded WebView component) would typically use the provided cache/persistence APIs in order to ensure that origin-bound data is preserved. This specification does not mandate particular mitigation strategies for dealing with size limitations (e.g., cache invalidation, storage reset), but merely recommends to follow best practices as used on the Open Web Platform.
+						</p>
 					</li>
+
 					<li>
-						<p>If a Reading System allows users to store persistent data, it must treat that data as
+						<p>
+							Reading Systems that support scripting and network access should also include
+							methods to notify the user that network activity is occurring and/or that allow them to disable it.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							If a Reading System allows users/scripts to store persistent data, it must treat that data as
 							sensitive. Scripts may save persistent data through cookies and DOM storage but Reading
 							Systems may block such attempts. Reading Systems that allow users to store data must ensure
 							they do not make that data available to other unrelated documents (e.g., ones that could be
 							spoofed). In particular, checking for a matching document identifier (or similar metadata)
-							is not a valid method to control access to persistent data.</p>
-						<p>Reading Systems that allow local storage should also provide methods for users to inspect,
+							is not a valid method to control access to persistent data.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							Reading Systems that allow local storage should also provide methods for users to inspect,
 							disable, and delete that data. The Reading System must destroy the data if the corresponding
-							EPUB Publication is deleted.</p>
+							EPUB Publication is deleted from its "bookshelf" if it implements one; otherwise it must destroy the
```

Comment:
This should probably be deleted. Unless we can specify what it means to "delete" content from a "bookshelf", this is about as meaningful as saying RSes should delete data when it should delete data.

Reply:
+1